hbase_storage/src/main/java/com/splicemachine/si/impl/region/HBaseTxnFinder.java

```diff
@@ -89,15 +92,81 @@ public void close() throws IOException {
         }
     }
 
+    private static class TxWithTimestampFilter extends FilterBase {
+
+        @Override
+        public boolean filterRowKey(Cell cell) {
```

Comment:
`filterRowKey` should only filter based on the rowkey, here you are looking at specific values. This is most likely called only once per row.

I'm guessing the current implementation wouldn't work on SPLICE_TXN, since we'd inspect the latest written cell (COMMITTED or ROLLEDBACK), see that it's not the ACTIVE state and filter it out, which would filter out the whole row.

Reply:
Thanks Daniel, I fixed it, now we simply return `false` in `filterRowKey()` method and I provided an implementation for `filterCell` instead, which returns `INCLUDE` for `ACTIVE` state.

Reply:
Looks good, I think you don't need to implement it at all, the implementation on the super class should be good too, but it's OK with me.